src/intersection-observer.js

```diff
@@ -208,6 +209,10 @@ export class IntersectionObserver extends Observable {
    * @private
    */
   flush_() {
+    if (!this.isElementInDoc_()) {
```

Comment:
Well if that check is expensive, I'll remove it. 
>We'd have to audit anywhere that's using an IntersectionObserver, but that's only <amp-ad>, <amp-a4a>, and <amp-iframe>. This is a much better choice, though.

We already did this and have the`destroy` function called in those elements. I add the check here for backup safety in cases like remove element during reparenting or cases where I didn't initiate the remove. But since the check is expensive, you're right, I don't want to check this everytime. I'll just remove it.